java/integTest/src/test/java/glide/cluster/CommandTests.java

```diff
@@ -556,4 +558,46 @@ public void time_with_route() {
                 "Time() result (" + serverTime[0] + ") should be greater than now (" + now + ")");
         assertTrue(Long.parseLong((String) serverTime[1]) < 1000000);
     }
+
+    @Test
+    @SneakyThrows
+    public void bgsave() {
+        var error = "Background save already in progress";
+        // bgsave may fail with given error and may keep failing on retries regardless of the
+        // timings/delays
+
+        var response = tryCommandWithExpectedError(() -> clusterClient.bgsave(false), error);
+        assertTrue(response.getValue() != null || response.getKey().startsWith("Background saving"));
+        Thread.sleep(2000); // next save call without delay will likely throw an error
+        response = tryCommandWithExpectedError(() -> clusterClient.bgsave(true), error);
+        assertTrue(response.getValue() != null || response.getKey().startsWith("Background saving"));
+        Thread.sleep(2000); // next save call without delay will likely throw an error
+
+        var transactionResponse =
```

Comment:
Can you not test these in ClusterTransaction as the last command to test or is there no guarantee that it may return an error?

Reply:
TBD, sometimes it works without errors

Reply:
We generally don't test transactions here I would add the test there.

Reply:
We can't test this in our common transaction IT unfortunately.

Reply:
Not in the common transaction can you not add it as a separate test but to the TransactionTests file?

Reply:
https://github.com/Bit-Quill/glide-for-redis/blob/main/java/integTest/src/test/java/glide/cluster/ClusterTransactionTests.java this file.